task-service/src/main/java/com/infodation/task_service/services/TaskServiceImpl.java

```diff
@@ -29,18 +28,33 @@
 
 @Service
 public class TaskServiceImpl implements ITaskService {
-
     private final TaskRepository taskRepository;
     private final ITaskCategoryService taskCategoryService;
     private final ITaskStatusService taskStatusService;
     private  final TaskServiceRepository taskServiceRepository;
+    private final TaskAssignmentRepository taskAssignmentRepository;
     private static final Logger log = LoggerFactory.getLogger(TaskServiceImpl.class);
 
-    public TaskServiceImpl(TaskRepository taskRepository, ITaskCategoryService taskCategoryService, ITaskStatusService taskStatusService, TaskServiceRepository taskServiceRepository) {
+    public TaskServiceImpl(TaskRepository taskRepository, ITaskCategoryService taskCategoryService, ITaskStatusService taskStatusService, TaskServiceRepository taskServiceRepository, TaskAssignmentRepository taskAssignmentRepository) {
         this.taskRepository = taskRepository;
         this.taskCategoryService = taskCategoryService;
         this.taskStatusService = taskStatusService;
         this.taskServiceRepository = taskServiceRepository;
+        this.taskAssignmentRepository = taskAssignmentRepository;
+    }
+
+    public void assignTaskToUser(TaskAssignmentDTO taskAssignmentDTO) {
+        try {
+            Task task = taskRepository.findById(taskAssignmentDTO.getTaskId())
+                    .orElseThrow(() -> new IllegalArgumentException("Task not found with ID: " + taskAssignmentDTO.getTaskId()));
+            UserTaskAssignment assignment = new UserTaskAssignment();
+            assignment.setUserId(taskAssignmentDTO.getUserId());
+            assignment.setTask(task);
```

Comment:
Using constructor with arguments for shorten code
```suggestion
            UserTaskAssignment assignment = new UserTaskAssignment(taskAssignmentDTO.getUserId(), task);
```